### PR TITLE
Jump Search in C

### LINKED ISSUE #2611: JumpSearch implementation in C
## 🚀 Feature

Jump sort algorithm implementation in C language.

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes

## Motivation

To add this new feature.

Please assign this feature to me under GSSOC'21.

### PR DESCRIPTION
**Pull request by nidh-ish**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Added Jump Search in C.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Fixes #2611

### CODE REVIEW

File: C/search/JumpSearch.c [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+	else {
+		printf("The element %d is not present in the array.\n", key);
+	}
+
+	return 0;
+}
+
+/* To run the program - 
+ *
+ * 	gcc JumpSearch.c -lm (Since, we have included math.h)
```
Comment (HarshCasper) [inline C/search/JumpSearch.c]: Don't use this

File: C/search/JumpSearch.c [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+ * 	./a.out
+ *
+ * The output of the program is -
+ *
+ * 	The element 92 is present at index 6 of the array.
+ *
+ * We can replace the value of key to search for different elements.
+ * Array can also be changed.
```
Comment (HarshCasper) [inline C/search/JumpSearch.c]: Give only sample I/O. Not instructions for how to run it or any other information

File: C/search/JumpSearch.c [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+	}
+
+	//We reuturn -1 if we did not find the key in the array
+	return -1;
+}
+
+int main() {
+
+	//Declaring the sorted array
+	int arr[] = {1, 23, 26, 35, 65, 89, 92, 112, 134, 157, 167, 202, 211, 267, 290, 297};
```
Comment (HarshCasper) [inline C/search/JumpSearch.c]: Make the input dynamic from the user

File: C/search/JumpSearch.c [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+		//the last index  of array + 1
+		if(jump >= size) {
+			jump = size;
+		}
+	}
+
+	//As the while loop ends, we are sure that the key, if present in the array
+	//is present between index initial and jump.
+	//
+	//We linearlly travere that interval to look for the key
```
Comment (HarshCasper) [inline C/search/JumpSearch.c]: Use multi-line comments

File: C/search/JumpSearch.c [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+	int jump = sqrt(size);
+
+	int initial = 0;
+
+	//Ensures that value of jump does not cross the size of the array.
+	//And also checks if the value at that jump is less than key or not.
+	while(jump < size && array[jump] <= key) {
+
+		//This keeps track of the lower bound of the interval
+		//in which key could be present
```
Comment (HarshCasper) [inline C/search/JumpSearch.c]: Use multi-line comments

File: C/search/JumpSearch.c [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+		initial = jump;
+
+		//Everytime we increase jump by sqrt(size of the array)
+		jump = jump + sqrt(size);
+
+		//If jump exceeds the size of the array, then we set jump as 
+		//the last index  of array + 1
```
Comment (HarshCasper) [inline C/search/JumpSearch.c]: Use multi-line comments

File: C/search/JumpSearch.c [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+ */
+int jumpSearch(int array[], int key, int size) {
+
+	//Size to jump
+	int jump = sqrt(size);
+
+	int initial = 0;
+
+	//Ensures that value of jump does not cross the size of the array.
+	//And also checks if the value at that jump is less than key or not.
```
Comment (HarshCasper) [inline C/search/JumpSearch.c]: Use multi-line comments

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by nidh-ish** [comment]
@HarshCasper Please check it.

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**